### PR TITLE
[irteus/irtmodel.l] Display warning when input angle-vector dimension > the number of joints

### PR DESCRIPTION
**Pull request by tkmtnt7000**
Related to https://github.com/sktometometo/jsk_robot/issues/131.

When directly inputting the angle-vector of a robot model, there was no error or warning when inputting a vector with more dimensions than the number of joints. The reason for this is that, in the implementation, the input vectors with more dimensions than the number of joints are ignored.

I think that not displaying warning messages is a little problematic for debugging or a little confusing (such as https://github.com/sktometometo/jsk_robot/issues/131)

- before
```lisp
3.irteusgl$ (send *pr2* :angle-vector #f(11.5 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 90.0))
#f(11.5 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0) ;; no error or warning
```
- after
```lisp
10.irteusgl$ (send *pr2* :angle-vector #f(11.5 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 90.0))
;; WARNING : Dimension of input vector: 18 > Dimension of joint-list: 17
#f(11.5 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0 0.0 -8.59437 0.0 -5.72958 0.0 0.0 0.0)
```

### CODE REVIEW

File: irteus/irtmodel.l [hunk 1]
```diff
@@ -821,6 +821,11 @@
               (angle-vector (instantiate float-vector (calc-target-joint-dimension joint-list))))
    "Returns angle-vector of this object, if vec is given, it updates angles of all joint. If given angle-vector violate min/max range, the value is modified."
    (let ((i 0))
+     (when vec
+       (when (> (length vec) (length joint-list))
```
Comment (Naoki-Hiraoka) [inline irteus/irtmodel.l:825]:
```suggestion
       (when (> (length vec) (apply #'+ (send-all joint-list :joint-dof)))
```

Comment (Naoki-Hiraoka) [reply irteus/irtmodel.l:825]: 複数自由度ある関節への対応です

File: irteus/irtmodel.l [hunk 2]
```diff
@@ -821,6 +821,11 @@
...
+         (warn ";; WARNING : Dimension of input vector: ~A > Dimension of joint-list: ~A~%"
+               (length vec)
+               (length joint-list))))
```
Comment (Naoki-Hiraoka) [inline irteus/irtmodel.l:828]:
```suggestion
               (apply #'+ (send-all joint-list :joint-dof))))
```

File: irteus/irtmodel.l (hunk 1 above)
Comment (tkmtnt7000) [reply irteus/irtmodel.l]: ありがとうございます．修正しました．

**Review by Naoki-Hiraoka — APPROVED** [review]

**Pull request merged**